### PR TITLE
Use TileMapLayer in demo dynamic_tilemap_layers since TileMap is deprecated

### PR DESCRIPTION
**Pull request by Yaxian**
Use `TileMapLayer` in demo dynamic_tilemap_layers since `TileMap` is deprecated

### CODE REVIEW

File: 2d/dynamic_tilemap_layers/project.godot [hunk 1]
```diff
@@ -30,6 +30,10 @@ window/size/viewport_height=495
 window/stretch/mode="canvas_items"
 window/stretch/aspect="expand"
 
+[dotnet]
+
+project/assembly_name="Dynamic TileMap Layers"
+
 [input]
```
Comment (Calinou) [inline 2d/dynamic_tilemap_layers/project.godot:37]:
```suggestion
[input]
```

File: 2d/dynamic_tilemap_layers/level/tile_map.gd [hunk 2]
```diff
@@ -21,26 +13,23 @@ func _process(delta: float) -> void:
 		if layer_alpha > 0.3:
 			# Animate the layer transparency.
 			layer_alpha = move_toward(layer_alpha, 0.3, delta)
-			set_layer_modulate(secret_layer, Color(1, 1, 1, layer_alpha))
+			set_self_modulate(Color(1, 1, 1, layer_alpha))
```
Comment (Calinou) [inline 2d/dynamic_tilemap_layers/level/tile_map.gd:16]:
```suggestion
			self_modulate = Color(1, 1, 1, layer_alpha)
```

File: 2d/dynamic_tilemap_layers/level/tile_map.gd [hunk 3]
```diff
@@ -21,26 +13,23 @@ func _process(delta: float) -> void:
...
 		else:
 			set_process(false)
 	else:
 		if layer_alpha < 1.0:
 			layer_alpha = move_toward(layer_alpha, 1.0, delta)
-			set_layer_modulate(secret_layer, Color(1, 1, 1, layer_alpha))
+			set_self_modulate(Color(1, 1, 1, layer_alpha))
```
Comment (Calinou) [inline 2d/dynamic_tilemap_layers/level/tile_map.gd:22]:
```suggestion
			self_modulate = Color(1, 1, 1, layer_alpha)
```

File: 2d/dynamic_tilemap_layers/level/tile_map.gd [hunk 4]
```diff
@@ -1,17 +1,9 @@
-extends TileMap
+extends TileMapLayer
 
 # You can have multiple layers if you make this an array.
-var secret_layer := 0
 var player_in_secret := false
 var layer_alpha := 1.0
```
Comment (Calinou) [inline 2d/dynamic_tilemap_layers/level/tile_map.gd:6]:
```suggestion
var layer_alpha := 1.0


```

File: 2d/dynamic_tilemap_layers/level/tile_map.gd [hunk 5]
```diff
@@ -21,26 +14,23 @@ func _process(delta: float) -> void:
...
 	else:
 		if layer_alpha < 1.0:
 			layer_alpha = move_toward(layer_alpha, 1.0, delta)
-			set_layer_modulate(secret_layer, Color(1, 1, 1, layer_alpha))
+			self_modulate = Color(1, 1, 1, layer_alpha)
 		else:
 			set_process(false)
 
+func _use_tile_data_runtime_update(_coords: Vector2i) -> bool:
+	return true
```
Comment (Calinou) [inline 2d/dynamic_tilemap_layers/level/tile_map.gd:29]:
```suggestion


func _use_tile_data_runtime_update(_coords: Vector2i) -> bool:
	return true


```

File: 2d/dynamic_tilemap_layers/level/tile_map.gd [hunk 6]
```diff
@@ -21,26 +14,25 @@ func _process(delta: float) -> void:
...
-func _use_tile_data_runtime_update(layer: int, _coords: Vector2i) -> bool:
-	return layer == secret_layer
+func _use_tile_data_runtime_update(_coords: Vector2i) -> bool:
+	return true
 
 
-func _tile_data_runtime_update(_layer: int, _coords: Vector2i, tile_data: TileData) -> void:
+func _tile_data_runtime_update(_coords: Vector2i, tile_data: TileData) -> void:
 	# Remove collision for secret layer.
 	tile_data.set_collision_polygons_count(0, 0)
```
Comment (Calinou) [inline 2d/dynamic_tilemap_layers/level/tile_map.gd:35]:
```suggestion


```

**Review by Calinou — APPROVED** [review]
Tested locally on 4.3.stable, it works as expected.

Thanks!

**Comment by KoBeWi** [comment]
Note that runtime update is no longer needed to disable collisions, because TileMapLayer has a property for that. The demo is no longer useful.
I opened an issue about it: #1163

**Pull request merged**